services/collaboration/src/events/broker.ts

```diff
@@ -0,0 +1,77 @@
+import amqp, { Connection, Channel } from 'amqplib/callback_api';
+import config from '../config';
+
+let channel: Channel | null = null;
+let connection: Connection | null = null;
+
+/**
+ * Function to connect to an existing RabbitMQ channel via RABBITMQ_URL
+ */
+export const getChannel = async (): Promise<Channel> => {
+    if (!channel) {
+        try {
+            await new Promise<void>((resolve, reject) => {
+                amqp.connect(config.BROKER_URL, (err, conn) => {
+                    if (err) {
+                        console.error('RabbitMQ connection error:', err);
+                        return reject(err);
+                    }
+
+                    connection = conn;
+
+                    conn.createChannel((err, ch) => {
+                        if (err) {
+                            console.error('RabbitMQ channel creation error:', err);
+                            return reject(err);
+                        }
+                        channel = ch;
+                        resolve();
+                    });
+                });
+            });
+        } catch (error) {
+            console.error('Error during RabbitMQ connection or channel access:', error);
+            throw error;
+        }
+    }
+
+    if (!channel) {
+        throw new Error('Failed to get the RabbitMQ channel.');
+    }
+
+    return channel;
+};
+
+/**
+ * Function to consume a message from a RabbitMQ queue
+ * @param queue
+ * @param onMessage
+ */
+export const consumeMessageFromQueue = async <T>(queue: string, onMessage: (message: T) => void): Promise<void> => {
+    try {
+        const ch = await getChannel();
```

Comment:
This creates a new channel each time we intend to consume a message from a queue, introducing unnecessary overhead. Wouldn't it be better to adapt the `MessageBroker` abstraction I made for the match service?